python-avd/pyavd/_eos_designs/structured_config/underlay/port_channel_interfaces.py

```diff
@@ -150,36 +147,69 @@ def port_channel_interfaces(self: AvdStructuredConfigUnderlayProtocol) -> list |
         # Now that validation is complete, we can make another pass at all l3_port_channels
         # (subinterfaces or otherwise) and generate their structured config.
         for l3_port_channel in self.shared_utils.node_config.l3_port_channels:
-            port_channel_interface = self._get_l3_port_channel_cfg(l3_port_channel)
-            append_if_not_duplicate(
-                list_of_dicts=port_channel_interfaces,
-                primary_key="name",
-                new_dict=port_channel_interface,
-                context=f"L3 Port-Channel interfaces defined under {self.shared_utils.node_type_key_data.key} l3_port_channels",
-                context_keys=["name", "peer", "peer_port_channel"],
-            )
+            self._set_l3_port_channel(l3_port_channel)
 
         # WAN HA interface for direct connection
-        if (port_channel_interface := self._get_direct_ha_port_channel_interface()) is not None:
-            append_if_not_duplicate(
-                list_of_dicts=port_channel_interfaces,
-                primary_key="name",
-                new_dict=port_channel_interface,
-                context="Port-Channel interface for WAN direct HA.",
-                context_keys=["name", "peer", "peer_interface"],
+        self._set_direct_ha_port_channel_interface()
+
+    def _set_l3_port_channel(
+        self: AvdStructuredConfigUnderlayProtocol, l3_port_channel: EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannelsItem
+    ) -> None:
+        """Set structured_configuration for one L3 Port-Channel."""
+        # build common portion of the interface cfg
+        interface = self._get_l3_common_interface_cfg(l3_port_channel)
+
+        interface_description = l3_port_channel.description
+        if not interface_description:
+            interface_description = self.shared_utils.interface_descriptions.underlay_port_channel_interface(
+                InterfaceDescriptionData(
+                    shared_utils=self.shared_utils,
+                    interface=l3_port_channel.name,
+                    peer=l3_port_channel.peer,
+                    peer_interface=l3_port_channel.peer_port_channel,
+                    wan_carrier=l3_port_channel.wan_carrier,
+                    wan_circuit_id=l3_port_channel.wan_circuit_id,
+                ),
+            )
+        interface._update(description=interface_description, peer_type="l3_port_channel", peer_interface=l3_port_channel.peer_port_channel)
```

Comment:
should we add `or None` here to be on the safe side?
```suggestion
        interface._update(description=interface_description, peer_type="l3_port_channel", peer_interface=l3_port_channel.peer_port_channel)
```

Reply:
hmm actually it can if someone gives a template that returns an empty string